doc/effects.md

```diff
@@ -1,268 +1,428 @@
+  (0) at the end of the reverse stage. During this time, the effect's progress is at 0%. If there
+  is no reverse stage, then this pause will still be inserted after the "at-max" pause if it's
+  present, or after the forward stage otherwise. In addition, the effect will now complete at
+  progress level of 0.
 
-This effect will change the base color of the paint, causing the rendered component to be tinted by
-the provided color.
+- **`repeatCount`** -- if greater than one, it will cause the effect to repeat itself the prescribed
+  number of times. Each iteration will consists of the forward stage, pause at max, reverse stage,
+  then pause at min (skipping those that were not specified).
 
-Usage example:
+- **`infinite`** -- if true, the effect will repeat infinitely and never reach completion. This is
+  equivalent to as if `repeatCount` was set to infinity.
+  
+- **`startDelay`** -- an additional wait time inserted before the beginning of the effect. This
+  wait time is executed only once, even if the effect is repeating. During this time the effect's
+  `.started` property returns false. The effect's `onStart()` callback will be executed at the end
+  of this waiting period.
+
+  Using this parameter is the simplest way to create a chain of effects that execute one after
+  another (or with an overlap).
+
+The effect controller returned by this factory constructor will be composited of multiple simpler
+effect controllers described further below. If this constructor proves to be too limited for your
+needs, you can always create your own combination from the same building blocks.
+
+In addition to the factory constructor, the `EffectController` class defines a number of properties
+common for all effect controllers. These properties are:
+
+- `.started` -- true if the effect has already started. For most effect controllers this property
+  is always true. The only exception is the `DelayedEffectController` which returns false while the
+  effect is in the waiting stage.
+
+- `.completed` -- becomes true when the effect controller finishes execution.
+
+- `.progress` -- current value of the effect controller, a floating-point value from 0 to 1. This
+  variable is the main "output" value of an effect controller.
+
+- `.duration` -- total duration of the effect, or `null` if the duration cannot be determined (for
+  example if the duration is random or infinite).
+
+
+### `LinearEffectController`
+
+This is the simplest effect controller that grows linearly from 0 to 1 over the specified
+`duration`:
 
 ```dart
-myComponent.add(
-  ColorEffect(
-    color: const Color(0xFF00FF00),
-    duration: 0.5,
-  ),
-);
+final ec = LinearEffectController(3);
+```
+
+
+### `ReverseLinearEffectController`
+
+Similar to the `LinearEffectController`, but it goes in the opposite direction and grows linearly
+from 1 to 0 over the specified duration:
+
+```dart
+final ec = ReverseLinearEffectController(1);
```

Comment:
I don't think you need the `final ec = ` part, since the variable isn't used or referred to?

Reply:
If I remove that part, Android Studio starts complaining that it is not a valid dart code:
<img width="398" alt="Screen Shot 2021-12-06 at 10 38 39 AM" src="https://user-images.githubusercontent.com/4231472/144902999-b41c57d6-bfe9-49e6-9025-e110cca26459.png">

With the variable assignment Android Studio is happy:
<img width="479" alt="Screen Shot 2021-12-06 at 10 39 57 AM" src="https://user-images.githubusercontent.com/4231472/144903167-be70cc3d-9461-4219-9aa3-95d87d4c813f.png">